src/mqt/bench/benchmark_generation.py

```diff
@@ -10,286 +10,206 @@
 
 from __future__ import annotations
 
+from enum import Enum, auto
 from importlib import import_module
-from pathlib import Path
-from typing import TYPE_CHECKING, Literal, overload
+from typing import TYPE_CHECKING
 
 from qiskit import generate_preset_pass_manager
+from qiskit.circuit import QuantumCircuit, SessionEquivalenceLibrary
 from qiskit.compiler import transpile
 from qiskit.transpiler import Target
+from typing_extensions import assert_never
 
-from .output import (
-    OutputFormat,
-    save_circuit,
-)
-from .targets.gatesets import get_target_for_gateset
+from .targets.gatesets import get_target_for_gateset, ionq, rigetti
 
 if TYPE_CHECKING:  # pragma: no cover
     from types import ModuleType
 
-    from qiskit.circuit import QuantumCircuit
     from qiskit.transpiler import Target
 
-from dataclasses import dataclass
 
-from qiskit.circuit import SessionEquivalenceLibrary
+class BenchmarkLevel(Enum):
+    """Enum representing different levels."""
 
-from .targets.gatesets import ionq, rigetti
+    ALG = auto()
+    INDEP = auto()
+    NATIVEGATES = auto()
+    MAPPED = auto()
 
 
-@dataclass
-class QiskitSettings:
-    """Data class for the Qiskit compiler settings."""
+# qubits → (nickname, N, a)
+SHOR_SIZE_TO_PARAMS = {
+    18: ("small", 15, 4),
+    42: ("medium", 821, 4),
+    58: ("large", 11777, 4),
+    74: ("xlarge", 201209, 4),
+}
```

Comment:
Hm. I actually had in mind to put all of this Shor-specific code into the Shor.py file and have it expose the same function as all the other benchmarks that just takes the circuit size.
That would isolate this special handling to the class where it is necessary and would decouple it from the benchmark generation itself, which would become more uniform.
It could be noted in the documentation of the Shor class, that this algorithm is only limited for certain numbers of qubits.